src/commit-split.ts

```diff
@@ -15,32 +15,67 @@
 import {Commit} from './graphql-to-commits';
 import {relative} from 'path';
 
-interface CommitSplitOptions {
+export interface CommitSplitOptions {
+  // Defaults to './'
   root?: string;
+  // Include empty git commits: each empty commit is included
+  // in the list of commits for each path.
+  includeEmpty?: boolean;
+  // rather than split by top level folder, split by these package
+  // paths (e.g. ["packages/pkg1", "python/pkg1"]). Commits that
+  // only touch files under paths not specified here are ignored.
+  packagePaths?: string[];
 }
 
 export class CommitSplit {
   root: string;
+  includeEmpty: boolean;
+  packagePaths?: string[];
   constructor(opts?: CommitSplitOptions) {
     opts = opts || {};
     this.root = opts.root || './';
+    this.includeEmpty = !!opts.includeEmpty;
+    if (opts.packagePaths) {
+      const paths: string[] = [];
+      for (let newPath of opts.packagePaths) {
+        newPath = newPath.replace(/[/\\]$/, '');
+        for (const exPath of paths) {
+          if (newPath.indexOf(exPath) >= 0 || exPath.indexOf(newPath) >= 0) {
+            throw new Error(
+              `Path prefixes must be unique: ${newPath}, ${exPath}`
+            );
+          }
+        }
+        paths.push(newPath);
+      }
+      this.packagePaths = paths;
+    }
   }
+
   split(commits: Commit[]): {[key: string]: Commit[]} {
     const splitCommits: {[key: string]: Commit[]} = {};
     commits.forEach((commit: Commit) => {
       const dedupe: Set<string> = new Set();
       for (let i = 0; i < commit.files.length; i++) {
         const file: string = commit.files[i];
-        const splitPath = relative(this.root, file).split(/[/\\]/);
+        const path = relative(this.root, file);
+        const splitPath = path.split(/[/\\]/);
         // indicates that we have a top-level file and not a folder
         // in this edge-case we should not attempt to update the path.
         if (splitPath.length === 1) continue;
-        const pkgName = splitPath[0];
-        if (dedupe.has(pkgName)) continue;
+        const pkgName = this.packagePaths
+          ? this.packagePaths.find(p => path.indexOf(p) >= 0)
```

Comment:
I might add a comment to this line, pointing out that "if packagePaths has been set, only track paths within this set".